source/simulator/solver.cc

```diff
@@ -806,7 +806,13 @@ namespace aspect
         // it in n_expensive_stokes_solver_steps steps or less.
         catch (SolverControl::NoConvergence)
           {
-            const unsigned int number_of_temporary_vectors = (parameters.include_melt_transport ? 100 : 50);
+            unsigned int number_of_temporary_vectors = parameters.stokes_gmres_restart_length;
+
+	    // use at least a restart length of 100 for melt models
+	    if (parameters.include_melt_transport &&
+		number_of_temporary_vectors < 100 )
+	      number_of_temporary_vectors = 100; 
```

Comment:
You can combine all of this into one statement like
```
  const unsigned int number_of_... = (parameters.include_melt_transport == false
                                                           ?
                                                           parameters.stokes_gmres_restart_length
                                                           :
                                                           std::max (parameters.stokes_gmres_restart_length, 100U));
```
This has the advantage that you can make the variable `const` right away.